app/jobs/inboxes/fetch_imap_email_inboxes_job.rb

```diff
@@ -3,7 +3,13 @@ class Inboxes::FetchImapEmailInboxesJob < ApplicationJob
 
   def perform
     Inbox.where(channel_type: 'Channel::Email').all.each do |inbox|
-      ::Inboxes::FetchImapEmailsJob.perform_later(inbox.channel) if inbox.channel.imap_enabled
+      next unless inbox.channel.imap_enabled
+
+      if ENV.fetch('AZURE_TENANT_ID', false) && inbox.channel.microsoft?
```

Comment:
What if the Microsoft channel is present, but the TENANT_ID is not configured?

Reply:
It will revert back to SMTP, which is how it worked before. Never tested multi-tenant, but I assume SMTP was working for that.